src/includes/dinit-iostream.h

```diff
@@ -0,0 +1,822 @@
+    int current_state() noexcept;
+
+    // Returns 1 when there is no set bits/vars otherwise 0
+    bool good() noexcept;
+
+    // Returns true or false if buffer_failbit is set or not
+    bool buffer_failure() noexcept;
+
+    // Returns 0 or POSIX errno based on posixio_errno value
+    int posixio_failure() noexcept;
+
+    // Resets failure bits/vars to false (or 0 about posixio_errno)
+    void clear() noexcept;
+
+    // Flush buffer of stream
+    // Non _nothrow variant may throw:
+    // dio::iostream_error on Buffer failure (buffer did not allocated at construction time)
+    // dio::posixio_error on POSIX I/O failure
+    ssize_t flush_nothrow() noexcept;
+    ssize_t flush();
+
+    // write() functions, Used for write many types
+    // Non _nothrow variants may throw:
+    // dio::iostream_error on Buffer failure (buffer did not allocated at construction time)
+    // dio::posixio_error on POSIX I/O failure
+    ssize_t write_nothrow(const char *msg) noexcept;
+    ssize_t write(const char *msg);
+
+    ssize_t write_nothrow(const char msg) noexcept;
+    ssize_t write(const char msg);
+
+    ssize_t write_nothrow(const std::string &msg) noexcept;
+    ssize_t write(const std::string &msg);
+
+    ssize_t write_nothrow(const int num) noexcept;
+    ssize_t write(const int num);
+
+    ssize_t write_nothrow(const unsigned num) noexcept;
+    ssize_t write(const unsigned num);
+
+    ssize_t write_nothrow(const long num) noexcept;
+    ssize_t write(const long num);
+
+    ssize_t write_nothrow(const unsigned long num) noexcept;
+    ssize_t write(const unsigned long num);
+
+    ssize_t write_nothrow(const endline &) noexcept;
+    ssize_t write(const endline &);
+
+    ssize_t write_nothrow(const flushbuf &) noexcept;
+    ssize_t write(const flushbuf &);
+
+    template <typename A, typename B, typename ...C> inline ssize_t write_nothrow(const A &a, const B &b, const C & ...c)
+            noexcept
+    {
+        if (!good()) return -1;
+        ssize_t ret1 = write_nothrow(a);
+        if (ret1 < 0) return -1;
+        ssize_t ret2 = write_nothrow(b, c...);
+        return (ret2 > 0) ? ret1 + ret2 : ret1;
+    }
+
+    template <typename A, typename B, typename ...C> inline ssize_t write(const A &a, const B &b, const C & ...c)
+    {
+        if (!good()) return -1;
+        ssize_t ret1 = write(a);
+        if (ret1 < 0) return -1;
+        ssize_t ret2 = write(b, c...);
+        return (ret2 > 0) ? ret1 + ret2 : ret1;
+    }
+
+    ~fileout() noexcept;
+};
+
+// Standard output "cout" on STDOUT_FILENO file descriptor
+extern fileout cout;
+// Standard error output "cerr" on STDERR_FILENO file desclogriptor
+extern fileout cerr;
+
+template <typename... arguments> ssize_t print(const arguments & ...args) noexcept
+{
+    return cout.write_nothrow(args...);
+}
+
+template <typename... arguments> ssize_t print_err(const arguments & ...args) noexcept
+{
+    return cerr.write_nothrow(args...);
+}
+
+/*
+ * filein
```

Comment:
should be `file_in` for consistency with existing style.

Reply:
Reverted to `istream`